### PR TITLE
Transaction History

### PR DESCRIPTION
**Pull request by simonmcl**
Changes:

History was in ascending order, oldest transaction first, changed to descending.
History was missing any send / receive involving FA1.2 tokens as they need to be searched for differently.
Added other types such as "Delegation", "Reveal", "Orignation" etc. to show users everything happening with their wallet.
Added extra attributes to model to enable parsing of contract data, display whether the transaction was successful or failed, what the new delegate address is etc.
Changed some values to optional to allow parsing for delegation and smart contract operations.

### CODE REVIEW

File: TezosKit/Conseil/ConseilClient.swift [hunk 1]
```diff
@@ -170,6 +180,24 @@ public class ConseilClient {
     networkClient.send(rpc, completion: completion)
   }
 
+	/// An RPC which fetches received transactions for an account, that came from smart contracts. For example if an account reiceved some FA1.2 tokens
```
Comment (keefertaylor) [inline TezosKit/Conseil/ConseilClient.swift:183]: nit: misspelling of 'received'

**Comment by keefertaylor** [comment]
Looks like you'll need to update the tests for the new `Transaction` format.

File: TezosKit/Conseil/ConseilClient.swift (hunk 1 above)
Comment (simonmcl) [reply TezosKit/Conseil/ConseilClient.swift]: I'll have you know, typos are my superpower

**Comment by simonmcl** [comment]
fixed tests, fixed typo, fixed indentation (ew spaces lol)

**Comment by keefertaylor** [comment]
@simonmcl CI is still failing?

**Pull request merged**